qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -286,7 +286,9 @@ def compose(self, other, qargs=None, front=False):
             z4[:, qargs] = z3
             pauli_list = PauliList(BasePauli(z4, x4, phase))
 
-        coeffs = np.kron(self.coeffs, other.coeffs)
+        # note: equivalent to `coeffs = np.kron(self.coeffs, other.coeffs)`
+        # since `self.coeffs` and `other.coeffs` are both 1d arrays.
+        coeffs = np.ravel(self.coeffs[:, np.newaxis] @ other.coeffs[np.newaxis])
```

Comment:
`np.multiply.outer(self.coeffs, other.coeffs).ravel()` might be worth a check in your benchmarks as well, if you're trying to eke out as much speed as possible.

Reply:
Thanks. Your suggestion is more comprehensive and the performance is almost same. I will use it.

`np.ravel(self.coeffs[:, np.newaxis] @ other.coeffs[np.newaxis])`
```
20, 4000: 1.356924046000131 sec
```

`np.multiply.outer(self.coeffs, other.coeffs).ravel()`
```
20, 4000: 1.3874290409999048 sec
```

Reply:
Hmmm, based on your benchmark I would have left yours in place, though perhaps it's just because of the statistical uncertainty in the `number=1` call in your `timeit`.  I tried a microbenchmark locally of just these lines and found `np.multiply.outer().ravel()` was between 20% and 2x faster than `x[:, newaxis] @ y[newaxis]` depending on the size of the problem - it seemed like it might be more cache efficient, perhaps because the arrays remain 1D so some offset calculations in inner-loop code could be elided.

Reply:
Thank you for testing the code locally. I see the `outer` is almost same as my code. https://github.com/numpy/numpy/blob/v1.22.0/numpy/core/numeric.py#L851-L936
It's interesting to observe the performance difference. Anyways, I like the comprehensive one with `outer`.
Thank you for your suggestion again.